### PR TITLE
Skirmish refactoring

### PR DESCRIPTION
**Pull request by Skleni**
Skirmish games are now handled in the same way as LAN games. Previously, they worked differently with the state (player positions, colors, teams, ...) being only implicitly stored in the UI controls. Now they use the same model (`SkirmishGameSettings`, formerly known as `SkirmishGame`) as LAN games, which makes the code cleaner and simpler.

Based on that, the starting position setup now also works for network games.

### CODE REVIEW

File: src/OpenSage.Game/Network/SkirmishManager.cs [hunk 1]
```diff
@@ -13,20 +13,61 @@ namespace OpenSage.Network
...
+
+        public abstract bool IsStartButtonEnabled();
+        public abstract Task HandleStartButtonClickAsync();
+
+        public virtual void Stop()
+        {
+        }
+    }
+
+    public sealed class LocalSkirmishManager : SkirmishManager
```
Comment (Skleni) [inline src/OpenSage.Game/Network/SkirmishManager.cs:40]: I added a `LocalSkirmishManager` class for non-network skirmish games (next to the existing implementations for Host and Client).

File: src/OpenSage.Game/Network/SkirmishManager.cs [hunk 2]
```diff
@@ -13,20 +13,61 @@ namespace OpenSage.Network
...
+
+        public override Task HandleStartButtonClickAsync()
+        {
+            Connection = new EchoConnection();
+            Settings.Status = SkirmishGameStatus.ReadyToStart;
+            return Task.CompletedTask;
+        }
+    }
+
+    public abstract class NetworkSkirmishManager : SkirmishManager
```
Comment (Skleni) [inline src/OpenSage.Game/Network/SkirmishManager.cs:59]: This is now the base class for the Host and Client skirmish managers (the new `LocalSkirmishManager` doesn't need the network stuff).

File: src/OpenSage.Mods.Generals/Gui/GameOptionsUtil.cs [hunk 3]
```diff
@@ -16,33 +18,29 @@ namespace OpenSage.Mods.Generals.Gui
 {
     public class GameOptionsUtil
```
Comment (Skleni) [inline src/OpenSage.Mods.Generals/Gui/GameOptionsUtil.cs:19]: This class now contains most of the logic shared between `SkirmishGameOptions` and `LanGameOptions`. It was there before, but was quite a mess, with half the logic existing there and the other half being in the UI Callbacks.

File: src/OpenSage.Game/Network/SkirmishGameSettings.cs [hunk 4]
```diff
@@ -2,18 +2,20 @@
 
 namespace OpenSage.Network
 {
-    public class SkirmishGame
+    public class SkirmishGameSettings
```
Comment (Skleni) [inline src/OpenSage.Game/Network/SkirmishGameSettings.cs:5]: I figured `SkirmishGameSettings` was a better name (`SkirmishGame` being rather ambiguos).

File: src/OpenSage.Game/Gui/Wnd/Controls/ComboBox.cs [hunk 5]
```diff
@@ -258,7 +258,7 @@ protected override void LayoutOverride()
                 0,
                 ClientSize.Height,
                 ClientSize.Width,
-                itemsHeight);
+                itemsHeight + 2);
```
Comment (feliwir) [inline src/OpenSage.Game/Gui/Wnd/Controls/ComboBox.cs:261]: Why was this required?

File: src/OpenSage.Game/Network/SkirmishGameSettings.cs (hunk 4 above)
Comment (feliwir) [reply src/OpenSage.Game/Network/SkirmishGameSettings.cs:5]: Agreed :)

**Review by feliwir — APPROVED** [review]

File: src/OpenSage.Game/Gui/Wnd/Controls/ComboBox.cs (hunk 5 above)
Comment (Skleni) [reply src/OpenSage.Game/Gui/Wnd/Controls/ComboBox.cs:261]: Not entirely sure...I added code to not draw ListBoxItems that are outside of the visible area of the ListBox. This lead to the last item not being drawn without these 2 added pixels...I guess there's some kind of border or padding involved.
